engine/class_modules/sc_death_knight.cpp

```diff
@@ -7710,15 +7710,16 @@ void death_knight_t::default_apl_frost()
   cooldowns -> add_action( "fireblood,if=buff.pillar_of_frost.remains<=8&buff.empower_rune_weapon.up" );
 
   // Pillar of Frost
-  cooldowns -> add_action( this, "Pillar of Frost", "if=cooldown.empower_rune_weapon.remains", "Frost cooldowns" );
+  cooldowns -> add_action( this, "Pillar of Frost", "if=cooldown.empower_rune_weapon.remains|!talent.breath_of_sindragosa.enabled", "Frost cooldowns" );
   cooldowns -> add_talent( this, "Breath of Sindragosa", "use_off_gcd=1,if=cooldown.empower_rune_weapon.remains&cooldown.pillar_of_frost.remains" );
   cooldowns -> add_action( this, "Empower Rune Weapon", "if=cooldown.pillar_of_frost.ready&!talent.breath_of_sindragosa.enabled&rune.time_to_5>gcd&runic_power.deficit>=10|target.1.time_to_die<20" );
   cooldowns -> add_action( this, "Empower Rune Weapon", "if=(cooldown.pillar_of_frost.ready|target.1.time_to_die<20)&talent.breath_of_sindragosa.enabled&runic_power>60" );
-
+  cooldowns -> add_action( this, "Empower Rune Weapon", "if=talent.icecap.enabled" );
+	
   // Cold Heart and Frostwyrm's Fury
   cooldowns -> add_action( "call_action_list,name=cold_heart,if=talent.cold_heart.enabled&((buff.cold_heart.stack>=10&debuff.razorice.stack=5)|target.1.time_to_die<=gcd)" );
-  cooldowns -> add_talent( this, "Frostwyrm's Fury", "if=(buff.pillar_of_frost.remains<=gcd|(buff.pillar_of_frost.remains<8&buff.unholy_strength.remains<=gcd&buff.unholy_strength.up))&buff.pillar_of_frost.up&azerite.icy_citadel.rank<=1" );
-  cooldowns -> add_talent( this, "Frostwyrm's Fury", "if=(buff.icy_citadel.remains<=gcd|(buff.icy_citadel.remains<8&buff.unholy_strength.remains<=gcd&buff.unholy_strength.up))&buff.icy_citadel.up&azerite.icy_citadel.rank>=2" );
+  cooldowns -> add_talent( this, "Frostwyrm's Fury", "if=(azerite.icy_citadel.rank<=1&buff.pillar_of_frost.up&(buff.pillar_of_frost.remains<=gcd|buff.unholy_strength.remains<=gcd|buff.icy_citadel.remains<=gcd&buff.icy_citadel.up))" );
```

Comment:
These two lines have the same conditions, making the behavior the same regardless of Icy Citadel ranks. Is that intended? Moreover, even though you check for buff.unholy_strength.remains<=gcd, you don't check whether the buff is up or not.

Reply:
yeah, both have basically the same interaction. was just a rewrite of the previous lines in a cleaner, easier to read fashion, in my opinion anyways. and yes, looks like checking for Unholy Strength being up should have been in there, i must have missed that interaction somehow. 
https://www.raidbots.com/simbot/report/6EUPF67Jf3oT87sgYs1hSS